src/server/storage.zig

```diff
@@ -373,9 +395,68 @@ pub const ChunkCompression = struct { // MARK: ChunkCompression
 			.uniform => {
 				ch.data.palette[0] = main.blocks.Block.fromInt(try reader.readInt(u32));
 			},
-			_ => {
-				return error.corrupted;
-			},
+		}
+	}
+
+	pub fn compressBlockEntityData(ch: *chunk.Chunk, comptime target: Target, writer: *BinaryWriter) void {
+		ch.blockPosToEntityDataMapMutex.lock();
+		defer ch.blockPosToEntityDataMapMutex.unlock();
+
+		if(ch.blockPosToEntityDataMap.count() == 0) return;
+
+		writer.writeEnum(BlockEntityCompressionAlgo, .raw);
+
+		var iterator = ch.blockPosToEntityDataMap.iterator();
+		while(iterator.next()) |entry| {
+			const index = entry.key_ptr.*;
+			const blockEntityIndex = entry.value_ptr.*;
+			const block = ch.data.getValue(index);
+			const blockEntity = block.blockEntity() orelse continue;
+
+			var tempWriter = BinaryWriter.init(main.stackAllocator);
+			defer tempWriter.deinit();
+
+			if(target == .toDisk) {
+				blockEntity.onStoreServerToDisk(blockEntityIndex, &tempWriter);
+			} else {
+				blockEntity.onStoreServerToClient(blockEntityIndex, &tempWriter);
+			}
+
+			if(tempWriter.data.items.len == 0) continue;
+
+			writer.writeInt(u16, @intCast(index));
+			writer.writeVarInt(usize, tempWriter.data.items.len);
+			writer.writeSlice(tempWriter.data.items);
+		}
+	}
+
+	pub fn decompressBlockEntityData(ch: *chunk.Chunk, comptime side: main.utils.Side, reader: *BinaryReader) !void {
+		if(reader.remaining.len == 0) return;
+
+		const compressionAlgo = try reader.readEnum(BlockEntityCompressionAlgo);
+		std.debug.assert(compressionAlgo == .raw);
```

Comment:
Why assert here but `try` in block data?

Reply:
Currently only raw data is handled and I want there to be an immediate crash here when someone tries to add a first compression algorithm and forgets to change it here.